clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -210,17 +210,19 @@ public CompletableFuture<Void> maybeAutoCommitAllConsumed() {
         return maybeAutoCommit(subscriptions.allConsumed());
     }
 
+    boolean canAutoCommit() {
+        return autoCommitState.isPresent() && !subscriptions.allConsumed().isEmpty();
+    }
+
     /**
-     * The consumer needs to send an auto commit during the shutdown if autocommit is enabled.
+     * Returns an OffsetCommitRequest of all assigned topicPartitions and their current positions.
      */
-    Optional<NetworkClientDelegate.UnsentRequest> maybeCreateAutoCommitRequest() {
-        if (!autoCommitState.isPresent()) {
-            return Optional.empty();
-        }
-
-        OffsetCommitRequestState request = pendingRequests.createOffsetCommitRequest(subscriptions.allConsumed(), jitter);
+    NetworkClientDelegate.UnsentRequest commitAllConsumedPositions() {
```

Comment:
Again, maybe make clear that this only creates the request. also, in other methods you use `AllConsumed` and drop the `Positions`, so maybe `createCommitAllConsumedRequest` ?

Reply:
good idea